src/redistributors/morpho/AccountFundsDistribution.sol

```diff
@@ -0,0 +1,73 @@
+// SPDX-License-Identifier: BUSL-1.1
+
+pragma solidity 0.8.23;
+
+import { Common } from "./Common.sol";
+
+/**
+ * @title Account Funds Distribution
+ * @dev   An abstract implementation for distributing across accounts within vaults based on their shares.
+ */
+abstract contract AccountFundsDistribution {
+    mapping(address vault => uint256 totalShares) private _totalShares;
+    mapping(address vault => uint256 pointsPerShare) private _pointsPerShares;
+    mapping(address vault => uint256 pointsLost) private _pointsLost; // Ignore this for now.
+
+    mapping(address vault => mapping(address account => int256 correction)) private _correctionPoints;
+    mapping(address vault => mapping(address account => uint256 shares)) private _shares;
+
+    /// @dev Distributes the `distributable_` across all shares of `vault_`.
+    function _distribute(address vault_, uint256 distributable_) internal {
+        if (distributable_ == 0) return;
+
+        uint256 totalShares_ = _totalShares[vault_];
+
+        if (totalShares_ == 0) revert Common.ZeroTotalShares();
```

Comment:
I don't think you can ever reach this error since `totalShares_` is `0` only when `distributable_` is `0`, so it would return early in the first condition.
We should keep it though to avoid dividing by `0` if we ever encounter this edge case where `distributable_` would be different than `0` but `totalShares_` would be equal to `0`.
May be worth adding a comment to explain it.

Reply:
You can reach this error if there are no shares deposited for a specific vault, and distribute is called with a non-zero amount, for that vault. Which is totally possible. It just means someone called distribute for:
- a non-existent vault
- an existing vault where no shared have yet been deposited